modules/core/src/main/scala/dev/guardrail/generators/ServerGenerator.scala

```diff
@@ -50,6 +50,10 @@ object ServerGenerator {
           for {
             resourceName <- formatTypeName(className.lastOption.getOrElse(""), Some("Resource"))
             handlerName  <- formatTypeName(className.lastOption.getOrElse(""), Some("Handler"))
+            authenticationEnabled = securitySchemes.nonEmpty && routes.exists {
```

Comment:
I think, it's not necessary to have `securitySchemes` down in the routes class, because there is no guarantee that all of them or any of them are used in routes. A route in the OAS contains security schemes names it requires, this information should be enough (in case we don't care about the schemes types and give a user the responsibility to implement them)